--- mkbootdisk.sparc	Sun May 20 11:24:26 2001
+++ mkbootdisk.sparc.new	Sat Mar  2 10:47:39 2002
@@ -1,6 +1,9 @@
 #!/bin/sh
 
-# mkbootdisk
+# mkbootdisk - not tested !
+# 03.02.2002 -- PLDized by:
+#
+# djrzulf <przemyslaw@knycz.net>
 #
 # 19.05.2001 -- updated by:
 #
@@ -15,7 +18,7 @@
 device=/dev/fd0
 
 unset kernel
-unset mkinitrdargs
+unset geninitrdargs
 unset verbose
 unset witheth
 
@@ -26,13 +29,13 @@
 PATH=/sbin:$PATH
 export PATH
 
-VERSION=2.0
+VERSION=2.0-PLD
 
 usage () {
 	cat >&2 <<EOF
-usage: `basename $0` [--version] [--noprompt] [--mkinitrdargs <args>] 
+usage: `basename $0` [--version] [--noprompt] [--geninitrdargs <args>] 
 	[--device <devicefile>] [--verbose -v] <kernel>
-	(ex: `basename $0` --device /dev/fd1 2.2.19)
+	(ex: `basename $0` --device /dev/fd0 2.2.20-10)
 EOF
 	exit $1
 }
@@ -43,9 +46,9 @@
 		shift
 		device=$1
 		;;
-	--mkinitrdargs)
+	--geninitrdargs)
 		shift
-		mkinitrdargs=$1
+		geninitrdargs=$1
 		;;
 	--help)
 		usage 0
@@ -146,7 +149,7 @@
 [ -n "$verbose" ] && echo "done."
 
 [ -n "$verbose" ] && echo -n "Creating initrd image... "
-/sbin/mkinitrd $mkinitrdargs $witheth --ifneeded $MOUNTDIR/initrd.img $kernel
+/sbin/geninitrd $geninitrdargs $witheth --ifneeded $MOUNTDIR/initrd.gz $kernel
 [ -n "$verbose" ] && echo "done."
 
 [ -n "$verbose" ] && echo -n "Setting up $LOADER... "
@@ -164,7 +167,7 @@
 	label=linux
 	root=$rootdev
 EOF
-[ -f $MOUNTDIR/initrd.img ] && echo "	initrd=/initrd.img" >> $MOUNTDIR/etc/$LOADER.conf
+[ -f $MOUNTDIR/initrd.gz ] && echo "	initrd=/initrd.gz" >> $MOUNTDIR/etc/$LOADER.conf
 
 cat >> $MOUNTDIR/etc/$LOADER.conf <<EOF
 image=/$KERNFILE
@@ -172,11 +175,11 @@
 	append="load_ramdisk=$LOADRAMDISK prompt_ramdisk=1"
 	root=/dev/fd0
 EOF
-[ -f $MOUNTDIR/initrd.img ] && echo "	initrd=/initrd.img" >> $MOUNTDIR/etc/$LOADER.conf
+[ -f $MOUNTDIR/initrd.gz ] && echo "	initrd=/initrd.gz" >> $MOUNTDIR/etc/$LOADER.conf
 chmod 644 $MOUNTDIR/etc/$LOADER.conf
 
 cat >> $MOUNTDIR/boot/message <<EOF
-Press <return> (or wait 10 seconds) to boot your PLD GNU/Linux system
+Press <return> (or wait 10 seconds) to boot your PLD/Linux system
 from $rootdev. If you want to use a rescue disk, type "rescue" now.
 
 EOF
